autoload/cmake/buildsys.vim

```diff
@@ -491,10 +477,8 @@ endfunction
 "         list of available build targets
 "
 function! s:buildsys.GetTargets() abort
-    if len(l:self.targets) == 0
-        call s:RefreshTargets()
-    endif
```

Comment:
Because the available targets may change from outside factors (e.g. user running CMake from the cli). `buildsys` doesn't have a local version of targets anymore and `s:fileapi.Reparse` is already checking if the index changed. I figured it would be easier and more correct to just let the fileapi handle update detection. If this turns out to be too much overhead this can be removed but since most of the time the cached version should be correct this should be OK.

Reply:
I think the overhead would indeed be negligible